### ISSUE TITLE
Code generation : server can't find any consumer when specified a wildcard

### ISSUE THREAD

**Issue by Thlr** [go-swagger/go-swagger#2267]
## Problem statement
I am working on a project which requires to re-implement a partial kubernetes api server and went with go-swagger to bootstrap the server.
Before I go any further, I am not much literate in openapi concepts and didn't want to dig into documentation so sorry if this is a non issue! 

I've been having an issue with endpoints consuming data. The api is specified with wildcards to accept any kind of data and that seems to be a problem for go-swagger as it can't find any consumer for the data I give it.
Once those wildcards are replaced with concrete types (e.g. application/json) the server works as intended.

## Swagger specification
For obvious reasons I am not going to paste the entirety of the k8s swagger spec. You can find it here : [swagger.json](https://github.com/kubernetes/kubernetes/blob/release-1.17/api/openapi-spec/swagger.json)

Let's focus on the `/api/v1/namespaces/{namespace}/bindings` endpoint:

```
...
    "/api/v1/namespaces/{namespace}/bindings": {
      "parameters": [
        {
          "description": "When present, indicates that modifications should not be persisted. An invalid or unrecognized dryRun directive will result in an error response and no further processing of the request. Valid values are: - All: all dry run stages will be processed",
          "in": "query",
          "name": "dryRun",
          "type": "string",
          "uniqueItems": true
        },
        {
          "description": "fieldManager is a name associated with the actor or entity that is making these changes. The value must be less than or 128 characters long, and only contain printable characters, as defined by https://golang.org/pkg/unicode/#IsPrint.",
          "in": "query",
          "name": "fieldManager",
          "type": "string",
          "uniqueItems": true
        },
        {
          "description": "object name and auth scope, such as for teams and projects",
          "in": "path",
          "name": "namespace",
          "required": true,
          "type": "string",
          "uniqueItems": true
        },
        {
          "description": "If 'true', then the output is pretty printed.",
          "in": "query",
          "name": "pretty",
          "type": "string",
          "uniqueItems": true
        }
      ],
      "post": {
        "consumes": [
          "*/*"
        ],
        "description": "create a Binding",
        "operationId": "createCoreV1NamespacedBinding",
        "parameters": [
          {
            "in": "body",
            "name": "body",
            "required": true,
            "schema": {
              "$ref": "#/definitions/io.k8s.api.core.v1.Binding"
            }
          }
        ],
        "produces": [
          "application/json",
          "application/yaml",
          "application/vnd.kubernetes.protobuf"
        ],
        "responses": {
          "200": {
            "description": "OK",
            "schema": {
              "$ref": "#/definitions/io.k8s.api.core.v1.Binding"
            }
          },
          "201": {
            "description": "Created",
            "schema": {
              "$ref": "#/definitions/io.k8s.api.core.v1.Binding"
            }
          },
          "202": {
            "description": "Accepted",
            "schema": {
              "$ref": "#/definitions/io.k8s.api.core.v1.Binding"
            }
          },
          "401": {
            "description": "Unauthorized"
          }
        },
        "schemes": [
          "https"
        ],
        "tags": [
          "core_v1"
        ],
        "x-kubernetes-action": "post",
        "x-kubernetes-group-version-kind": {
          "group": "",
          "kind": "Binding",
          "version": "v1"
        }
      }
    },
...
```
## Steps to reproduce
- Generate the server with the spec (no need to configure consumers, default ones work fine)
- Implement the BearerTokenAuth authorization function (or don't? can't remember the default one)
```
 api.BearerTokenAuth = func(token string) (interface{}, error) {
    return "", nil 
}   
```
- POST some json on `/api/v1/namespaces/mynamespace/bindings` with some bearer token in the header

### Expected behavior
Response : `operation createCoreV1NamespacedBinding has not yet been implemented`

### Got
```
{
    "code": 500,
    "message": "no consumer registered for application/json"
}
```

In the spec, change
```
      "post": {
        "consumes": [
          "*/*"
        ],
```
to
```
      "post": {
        "consumes": [
          "application/json"
        ],
```
and it should work as intended.

Thank you for you help and for this fantastic tool.

## Environment
swagger version: v0.23.0
go version: 1.14
OS:  Linux (Manjaro)

**Comment by casualjim** [comment]
The question is which consumer do you want to take with '*/*' ?
For content types we essentially parse the content type as mime header and then chop off all the arguments. This gives us the base which becomes a simple map lookup. 
So if you just want 1 consumer to be the default, I think you can get away with registering a consumer on your api and setting the key to "*/*"

**Comment by Thlr** [comment]
Thank you for you response, that might just do. Excuse my lack of swagger expertise but how am I supposed to do that exactly? I added  this line : ` api.RegisterConsumer("/", runtime.JSONConsumer()) ` to my configure_xxx.go but it didn't seem to resolve my problem.
Btw, a json consumer is already defined (api.JSONConsumer is set)

**Comment by casualjim** [comment]
The wildcard mime type is `*/*` but you should enumerate all the content types you want to support, and set the appropriate consumers.

**Comment by Thlr** [comment]
Bear in mind that my objective here is not to change the swagger.json, but rather adapt to it. I have set `api.RegisterConsumer("application/json", runtime.JSONConsumer())` but it also doesn't lead to anything.
If you think I really should change the json to enumerate the content types then I'll stick to that solution but it still bugs me that the server answers that it couldn't find any consumer for the content when, in fact, I have set said consumer.

**Comment by casualjim** [comment]
The server uses the `Accept` header provided by the client to pick the right serializer, but defaults to json.

Even if you were to support `*/*` and then enumerate yourself, you'd still have to work out the types.
You'd still only be able to support a finite set of content types. And you'd have to figure out some way for the clients to signal which data format they want back.

**Comment by casualjim** [comment]
The server uses the `Accept` header provided by the client to pick the right serializer, but defaults to json.

Even if you were to support `*/*` and then enumerate yourself, you'd still have to work out the types.
You'd still only be able to support a finite set of content types. And you'd have to figure out some way for the clients to signal which data format they want back.

**Comment by casualjim** [comment]
The server uses the `Accept` header provided by the client to pick the right serializer, but defaults to json.

Even if you were to support `*/*` and then enumerate yourself, you'd still have to work out the types.
You'd still only be able to support a finite set of content types. And you'd have to figure out some way for the clients to signal which data format they want back.

**Comment by casualjim** [comment]
The server uses the `Accept` header provided by the client to pick the right serializer, but defaults to json.

Even if you were to support `*/*` and then enumerate yourself, you'd still have to work out the types.
You'd still only be able to support a finite set of content types. And you'd have to figure out some way for the clients to signal which data format they want back.

**Comment by Thlr** [comment]
Yes that makes sense, wildcard or not, I still got to work out which type I consume, and which one I produce. And honestly manually setting the consumers in the json isn't to big of a hassle, I can live with that.
The producer part is set properly and doesn't seem to be a problem, also it has already proven to be functioning well.

What I don't get though, is how swagger handles mime types. I sent a few test requests : when I set no Content-Type header, it can't find what consumer it should pick. When I set it to "application/json", or any other "real" type, same story. When I set it to "*/*" though and set the empty consumer to runtime.JSONConsumer(), everything works as expected!

Indeed, thank you very much for your time, but I just want to understand how go swagger handles these types :) If I'm forgetting some elements in my issue to make it clear, please let me know, I haven't done this many times.
I am very confusion.
